src/cache.ts

```diff
@@ -0,0 +1,64 @@
+import * as VError from 'verror'
+const debug = require('./debug').extend('cache')
+
+interface Config {
+  name: string
+  existsFn: (url: string) => Promise<boolean>
+  storeFn: (url: string, data: object) => object
+  retrieveFn: (url: string) => object
+  removeFn: (url: string) => object
+}
+class Cache {
+  name: string
+  debug: any
+  existsFn: (url: string) => Promise<boolean>
+  storeFn: (url: string, data: object) => object
+  retrieveFn: (url: string) => object
+  removeFn: (url: string) => object
+
+  static isValidConfig (x: any): x is Config {
```

Comment:
why `(x: any): x is Config` and not `(x: Config): boolean`?

Reply:
this is a type-guard, if we use this in a `if` condition, typescript will assume the type of the parameter to be `Config` inside the block of that `if` condition